content/oci/oci.go

```diff
@@ -526,6 +512,39 @@ func (s *Store) GC(ctx context.Context) error {
 	return nil
 }
 
+// gcIndex reloads the index and updates metadata. Information of untagged blobs
+// are cleaned and only tagged blobs remain.
+func (s *Store) gcIndex(ctx context.Context) error {
+	refMap := s.tagResolver.Map()
+	s.tagResolver = resolver.NewMemory()
+	s.graph = graph.NewMemory()
+	for ref, desc := range refMap {
+		if ref == desc.Digest.String() {
```

Comment:
Just realized, for the following graph in the OCI store,

```mermaid
graph TD
    Referrer --> Manifest
```
Where `Manifest` is tagged while `Referrer` is not tagged.
If we run `GC()`, will `Referrer` be deleted?

Reply:
Updated our implementation, so that now when `manifest` is tagged and `referrer` is not tagged, the referrer will not be deleted during `GC`. The rule is that nodes reachable (by referrer and successor relation) from tagged manifests will not be deleted.